src/main/java/DateReader.java

```diff
@@ -0,0 +1,115 @@
+public class DateReader {
+    protected String date;
+
+    public DateReader(String date) {
+        this.date = date;
+    }
+    
+    public static String readDate(DateReader dr) throws DateDoesNotExistException {
+    	String[] arr = dr.date.split("/");
+
+    	String dd = arr[0];
+    	String mm = arr[1];
+    	String yy = arr[2].split(" ")[0];
+        String time = arr[2].split(" ")[1];
+
+        if (Integer.parseInt(dd) > 31 || Integer.parseInt(dd) < 1) {
+            throw new DateDoesNotExistException("Day does not exist!");
+        }
+
+		if (Integer.parseInt(mm) > 12 || Integer.parseInt(mm) < 1) {
+			throw new DateDoesNotExistException("Month does not exist!");
+		}
+
+		if (Integer.parseInt(time) < 0 || Integer.parseInt(time)%100 > 59) {
+			throw new DateDoesNotExistException("Time does not exist!");
+		}
```

Comment:
Was this indentation intended? If not might get confusing for a regular reader. I like how you broke up your long conditions into separate conditions.